protos/table.proto

```diff
@@ -153,6 +153,7 @@ message Manifest {
   // If this value is 0 then there are no blob fields.
   uint64 blob_dataset_version = 17;
 
+  repeated string reference_paths = 18;
```

Comment:
I was originally thinking this is a map (technically something like repeated tuple of string, string in protobuf). The reason is that if we use an index to reference the base path, we will run into the issue that we can never remove a reference path? Since that will change the position of the path_base_index. What do you think?

Reply:
> I was originally thinking this is a map (technically something like repeated tuple of string, string in protobuf). The reason is that if we use an index to reference the base path, we will run into the issue that we can never remove a reference path? Since that will change the position of the path_base_index. What do you think?

I was using a map originally. But I encountered the issue that who could give the path key and be responsable that it won't conflict. Let users do that seems not reasonable. If let system decides I think an integer would be enough.

You are right about the remove case. I think we could use a `HashMap<u32, String>` as the base_paths type. What do you think.

Reply:
> Also I think this field should be called `base_paths`. "reference" is specific to cloning, but this "base" concept can be used for other cases like multi-tier storage.

Agree.
Actually I was on the fence about choosing base_paths or reference_paths

Reply:
> I think we could use a HashMap<u32, String> as the base_paths type. What do you think.

sounds good! Can we use a list of entries instead of map here? I think that is in general more backwards compatible than a map, like

```
repeated BasePath base_paths = 18;

message BasePath {
  uint32 id,
  string name,
  string path,
}
```

In this way, we can also assign a meaningful `name` to the base path. For the clone us case, this can just be for example the `ref_name`

Reply:
Totally agree with the BasePath

I think a map in manifest would be better for generating paths, or we will scan the paths to get the right one. The impact might be minor for now, but it feels unnecessary to introduce this overhead. What do you think?